### ISSUE TITLE
Feature Request: Mail Folder Widget

### ISSUE THREAD

**Issue by rodzam** [k9mail/k-9#1600]
It would be terrific if K9-Mail would incorporate an folder view widget, similar to the one the GMail (and other e-mail apps) have. I came across MailListWidget for K9 (http://maillistwidgetk9.lombardi.mobi/ since the Google Play page no longer seems to work) but it no longer seems to work. Email Widget (https://play.google.com/store/apps/details?id=de.foobarsoft.emailwidget) is also supposed to work, but is apparently limited to the Unified Mailbox and doesn't provide important information like whether the message is read or unread.

Here's an example from GMail (the widget is in the middle of the screenshot on the right side):
![image](https://cloud.githubusercontent.com/assets/3299186/18255812/5c9b2dae-737b-11e6-8fc9-92c2331f540d.png)

Thank you for a terrific app, and your consideration.

**Comment by ahmadaamer6** [comment]
@rodzam Try Dashclock (https://play.google.com/store/apps/details?id=net.nurik.roman.dashclock&hl=en) or Better Dashclock (https://play.google.com/store/apps/details?id=com.betterapps.dashclock&hl=en) w/ K-9 Mail DashClock Extension (https://play.google.com/store/apps/details?id=de.cketti.dashclock.k9) for a good widget.

**Comment by Denis1990** [comment]
I would really like to help implement this feature as i find it very usefull.
I have done some preliminary work and created this [commit](https://github.com/Denis1990/k-9/commit/c863e32e4897164d58038b596b76856498fb5e77)

**Comment by helmo** [comment]
There already is an 'UnreadWidget' included which lets you select one of your inboxes. 
Can't that one be extended to allow choosing any folder?

+1 for having such a widget.

**Comment by Denis1990** [comment]
@helmo I have implemented that same functionality for this widget as well.
Also i've done work to implement the basic functionality of the widget that is to view the messages as a list on the home screen.
I have opened a thread in [google groups](https://groups.google.com/forum/#!topic/k-9-dev/B2I3RrsiQfY) to ask questions.

**Comment by steffchep** [comment]
Yes, please, a widget listing the mails, rather than just show the number of unread, like most builtin mail clients have, as well as BlueMail/Aquamail! I, too used MailListWidget, and it worked fine until I had to reinstall everything after upgrading to Lollipop....
The fact that this feature is missing is the reason I moved from K9 to another app.... the ONLY reason.

**Comment by ascheman** [comment]
+1

**Comment by ascheman** [comment]
Somehow comments get lost???

I still would like to vote +1 for this enhancement!

**Comment by helmo** [comment]
Your "+1" comment did make it into my mailbox as a notification.

**Comment by philipwhiuk** [comment]
Please use GitHub reactions not comments. Comments just saying +1 are unhelpful when trying to see useful feedback.
